docs/basic-dsa/array/Searching-algorithm.md

```diff
@@ -0,0 +1,245 @@
+---
+id: arrys-Searching Algorithms-in-dsa
+title: Arrays - Searching Algorithms in DSA
+sidebar_label: Searching Algorithms
+sidebar_position: 2
+description:Searching Algorithms are methods used to find a specific value or condition within a collection of data. They are fundamental in computer science and software development, enabling efficient data retrieval from various data structures. This tutorial will cover two primary searching algorithms: Linear Search and Binary Search.
+tags: [dsa, arrays, searching algorithms]
+
+
+<AdsComponent />
+Searching Algorithms are methods used to find a specific value or condition within a collection of data. They are fundamental in computer science and software development, enabling efficient data retrieval from various data structures. This tutorial will cover two primary searching algorithms: Linear Search and Binary Search.
+
+##Linear Search
+Linear search is the simplest searching algorithm, which checks every element in a list until it finds the target value.
+
+##Algorithm
+1. Start from the first element of the array.
+2. Compare the current element with the target value.
+3. If they match, return the index of the current element.
+4. If not, move to the next element.
+5. Repeat steps 2-4 until the target is found or the end of the array is reached.
+6. If the target is not found, return -1. 
+ 
+ ##Pseudo Code
+ procedure binarySearch(A : sorted list of items, target)
+    left = 0
+    right = length(A) - 1
+    while left <= right do
+        mid = (left + right) / 2
+        if A[mid] == target then
+            return mid
+        else if A[mid] < target then
+            left = mid + 1
+        else
+            right = mid - 1
+    end while
+    return -1
+end procedure
```

Comment:
replace your code 

```
procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```

with

````md
```
procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```
````